File: dotnet/TestSnippets.Examples/Behavioural/SampleBehaviouralTest.cs
using System;
using Xunit;

namespace PublicDomain.TestSnippets.Behavioural
{
    public class SampleBehaviouralTest: ATestBase<SampleBehaviouralTest>
    {
        [Fact]
        public void DemonstrationOfSteps()
        {
            When().Time().Passes(TimeSpan.FromDays(2));
            Then().SpecialCaseShouldBeFulfilled();
        }

        /// <summary>
        /// This method is an example for a test-specific step.
        /// </summary>
        private void SpecialCaseShouldBeFulfilled() {
            Assert.True(true);
        }
    }
}

File: cpp/src/test/examples/behavioural/SampleBehaviouralTest.cpp
#include "ATestBase.h"

namespace publicdomain::testsnippets::examples {

/**
 * \link SampleBehaviouralTest \endlink shows a possible usage of
 * \link ATestBase \endlink.
 */
class SampleBehaviouralTest : public ATestBase<SampleBehaviouralTest> {
public:
  /**
   * This method is an example for a test-specific step.
   */
  void specialCaseShouldBeFulfilled() { EXPECT_TRUE(true); }
};

/**
 * This test case is an example on how steps would be used.
 */
TEST_F(SampleBehaviouralTest, demonstrationOfSteps) {
  when().time().passes(std::chrono::hours(20));
  then().specialCaseShouldBeFulfilled();
}

} // namespace publicdomain::testsnippets::examples
